Dockerhub image push second set of fixes.

NlightNFotis opened the pull request:
This should fix docker complaining about invalid reference format passed to --tag
during `docker build`.

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

NlightNFotis committed:
Add quotes before tag name.

This should fix docker complaining about
invalid reference format passed to --tag
during `docker build`.

hannes-steffenhagen-diffblue commented on .github/workflows/release-packages.yaml:
```diff
@@ -223,7 +223,7 @@ jobs:
         run: echo "IMAGE_TAG=diffblue/cbmc:${{ steps.get_release_info.outputs.tag_name }}" >> $GITHUB_ENV
       - name: Build docker image
-        run: docker build -t $IMAGE_TAG .
+        run: docker build -t "$IMAGE_TAG" .
```
🚫 What's it explaining about right now? What's the value of `$IMAGE_TAG` here? Image tags should be easy to reference by developers so shouldn't contain things like spaces or other weirdness

NlightNFotis replied:
I think that might have been a mistake from my part as well when I made the release. Because of the release version action we are using, and the release names I chose which were kind of elaborate, like `cbmc-5.21.2-exp`, it might be using the whole tag as a name.

NlightNFotis replied:
I mean the `" "` should certainly help in any case, but part of it was the release name I chose made the tag name more convoluted than it is allowed to be.

NlightNFotis committed:
Isolate numeric version for image tag.

Previously, we were using the full release name
for generating a dockerhub image name, which if
it contained any characters like `-` would be creating
an issue.

This is now performing some preprocessing to isolate
the numeric part of the cbmc version (a string of the
form `cbmc-5.20.2`) so that it should be able to be
used for a tag name without any issues.

NlightNFotis committed:
Add some debug output for the tag name

... from the `get_release_info` github action
so that we make sure the output of it is correct
so long as the tag name is concerned.

piotr-grabalski commented on .github/workflows/release-packages.yaml:
```diff
@@ -221,10 +221,13 @@ jobs:
         run: docker build -t "$IMAGE_TAG" .
       - name: Push docker image to DockerHub
         run: |
           docker login --username=dbcicprover --password=${{ secrets.DOCKERHUB_ACCESS_DB_CI_CPROVER }}
```
Would there be any overheads of keeping username in github secrets too? I think it might be easier to maintain if we ever change the username if it is (or is going to be) used in multiple places. It would make the credentials more secure too.

NlightNFotis replied:
I asked Peter about it, but he said that this doesn't need to be a secret.

piotr-grabalski replied:
OK, do you think you will need this in any other place so that it would be worth of keeping it as a define somewhere above?

NlightNFotis replied:
I don't think it's going to be needed anywhere else to be honest.

And if it is, I can do the refactoring later when there is evidence it's needed.

thomasspriggs replied:
The benefit of keeping the username as a secret would be that forks of CBMC could supply different secrets rather than diverge the code. However it shouldn't make any difference to us.
